src/entry_stream_stage.rs

```diff
@@ -59,9 +59,38 @@ impl EntryStreamStage {
     ) -> Result<()> {
         let timeout = Duration::new(1, 0);
         let entries = ledger_entry_receiver.recv_timeout(timeout)?;
-        entry_stream.stream_entries(&entries).unwrap_or_else(|e| {
-            error!("Entry Stream error: {:?}, {:?}", e, entry_stream.socket);
-        });
+
+        for entry in &entries {
+            if entry.is_tick() && entry_stream.queued_block.is_some() {
+                let queued_block = entry_stream.queued_block.as_ref();
+                let block_tick_height = queued_block.unwrap().tick_height;
+                let block_id = queued_block.unwrap().id;
+                entry_stream
+                    .emit_block_event(block_tick_height, block_id)
+                    .unwrap_or_else(|e| {
+                        error!("Entry Stream error: {:?}, {:?}", e, entry_stream.output);
+                    });
+                entry_stream.queued_block = None;
+            }
+            entry_stream.emit_entry_event(&entry).unwrap_or_else(|e| {
+                error!("Entry Stream error: {:?}, {:?}", e, entry_stream.output);
+            });
+            // This logic assumes that there is always at least one transaction (the leader's own vote)
```

Comment:
I don't quite understand this comment?  Where's the assumption exactly?

Reply:
The assumption is implemented by the `!entry.is_tick()` check.
But looking at this further, I realize it's not really necessary.